### ISSUE TITLE
SaeAssignment::row_block_dim is dense in K_atoms — large-K modes need active-set row layout

### ISSUE THREAD

**Issue by SauersML** [SauersML/gam#643]
## Summary

`SaeAssignment::row_block_dim()` at `src/terms/sae_manifold.rs:1190-1191` returns `self.k_atoms() + self.total_coord_dim()`, so the per-row local Hessian `A_i = H_tt^(i)` is `(K_atoms + Σ_k d_k) × (K_atoms + Σ_k d_k)` regardless of assignment type. The assembly at `:1698` allocates `Array2::<f64>::zeros((q, p))` and at `:1727` `local_jac.fill(0.0)` — fully dense in atom count for every row.

For sparse-assignment variants this is wasted work:

- **Softmax / IBPMap** (`:2420-2425`, `:2435-2442`) — genuinely dense across all `K_atoms` per row, so the dense layout is faithful to the math.
- **JumpReLU** (`:2452-2455`) — skips computation for `logits[col] ≤ threshold` via `continue`, but the dense `(q × p)` allocation, the dense `local_jac.fill(0.0)`, and the dense `q × q` `A_i` Cholesky / factor all still happen at the full `K_atoms` dimension.

So JumpReLU computes nothing for inactive atoms but still allocates and factorises as if they were active. For `K_atoms = 10⁵` and per-row active fraction ~1%, the cost of the sparse-mode path is ~99% bookkeeping on zeros.

## Why it matters

Every per-row Cholesky factor `A_i^{-1}` materialised at `(K_atoms + coord_dim)²` is O(K_atoms³) flops per row, O(K_atoms²) memory per row. At biobank `N = 10⁵`–`5·10⁵` and `K_atoms = 10⁴` upward, this is unworkable even if the dense block is mostly zero. Sparse-assignment variants were presumably introduced precisely because dense softmax does not scale — but the row representation makes them pay the dense cost anyway.

## What is needed

An active-set row layout that JumpReLU / TopK / sparsemax / hard-concrete variants can populate:

```rust
struct ActiveRow {
    active_logit_cols: SmallVec<[u32; 16]>,
    active_coord_blocks: SmallVec<[CoordBlockId; 8]>,
    local_dim: usize, // q_active = active_logit_cols.len() + Σ active coord dims
    local_jac: SmallDense, // (q_active × p)
    local_hessian: SmallDense, // (q_active × q_active)
    gt: SmallVec<[f64; 32]>,
}
```

`row_block_dim()` becomes per-row instead of a global constant. The arrow-Schur per-row Cholesky factor then has `q_active² ≪ K_atoms²` cost. The dense softmax path keeps the current layout as a special case where `q_active = K_atoms`.

This is independent of the Kronecker exploit on the beta side (#293): they compose multiplicatively — sparse rows × Kronecker scatter/gather is what gets SAE to biobank `K_atoms`.

## Scope reality

Tests today exercise only `K_atoms ∈ {1, 2}` (`tests/sae_manifold_sphere_torus_topologies.rs:198-267`, `bug_hunt_latent_sae_gated.rs`). At those sizes the dense-q allocation is invisible. The motivation is forward-looking; the change is mechanical refactoring of the `ArrowRowBlock` representation that SAE assembly produces, not a math change.

This also has knock-on impact on `ArrowSchurSystem::d` (currently a global scalar at `src/solver/arrow_schur.rs:756-895`). Variable per-row `d` is a real structural change to the arrow-Schur API — needs an `Arc<[usize]>` of row dims or a max-d + active-mask split. Worth scoping as a multi-PR effort.

Refs: row_block_dim at `src/terms/sae_manifold.rs:1190-1191`; allocation at `:1698`, `:1727`; JumpReLU sparsity at `:2452-2455`; Softmax dense at `:2420-2425`.

**Comment by SauersML** [comment]
Implemented active-set row layout for SAE-manifold JumpReLU mode and migrated ArrowSchurSystem from scalar d to per-row dims.

## Changes

### arrow_schur.rs — Variable per-row d (atomic scalar → per-row)
- Added `row_dims: Arc<[usize]>` and `row_offsets: Arc<[usize]>` to `ArrowSchurSystem` and `ArrowFactorCache`. `sys.d` becomes `max_d` (upper bound for scratch buffers).
- New constructor `ArrowSchurSystem::new_with_per_row_dims(per_row_dims, k)` for heterogeneous SAE rows.
- All solve paths (direct, sqrt-BA, InexactPCG, streaming, IFT predictors, evidence gradient, damped quadratic model) updated to use `row_dims[i]` / `row_offsets[i]` instead of the global scalar.
- `SaeKroneckerRows` drops the fixed `q` field; `apply_l` / `apply_l_t` derive per-row `q_i = local_jac[row].len() / p`.
- `StreamingArrowSchur` updated with `row_dims` + `row_offsets` + per-row `validate_row`.
- All helper functions (`sys_htbeta_apply_row`, `schur_matvec`, `JacobiPreconditioner`, `reduced_rhs_beta`, gradient dot-step, damped model reduction) use `row.htt.nrows()` per row.

### sae_manifold.rs — Active-set row layout for JumpReLU
- New `SaeRowLayout` struct: per-row active atom sets, compact→full-q expansion map.
- `assemble_arrow_schur` (now `&mut self`) computes JumpReLU active sets and allocates the system with `new_with_per_row_dims`. JumpReLU rows pay only `q_active_i² ≪ K_atoms²` per-row Cholesky cost.
- Active-set local Jacobian filled only for active atoms (logit > threshold), no allocations for inactive atoms.
- `apply_newton_step_impl` expands compact `delta_t` back to full-q via `SaeRowLayout::expand_row` before updating logits and coordinates.
- `solve_newton_step` changed to `&mut self`.
- Dense Softmax / IBPMap unchanged — keeps full-q layout as the special case.

### Caller updates
- `evidence.rs`, `persistent_warm_start.rs`, `latent_inner.rs`, `arrow_schur_convergence_check.rs` updated to use `cache.delta_t_len()`, `cache.row_dims[i]`, `cache.row_offsets[i]` instead of `n * d` / `i * d`.
- Test fixtures changed to `let mut f = build_fixture(...)`.

### Complexity gain
For JumpReLU at active fraction α:  
- Per-row Cholesky: O((αK)³) vs O(K³) — factor 1/α³  
- Per-row Jacobian: O((αK) · p) vs O(K · p) — factor 1/α  
- Composes multiplicatively with Kronecker β-side (#293).
